lib/foreman/force_ssl.rb

```diff
@@ -0,0 +1,46 @@
+module Foreman
+  class ForceSsl
+    UNATTENTED_PATHS = %r{^/(unattended|userdata)/}
+
+    def self.allowed_http_actions
+      @allowed_http_actions ||= []
+    end
+
+    def self.add_allowed_http_action!(action)
+      allowed_http_actions << action
+    end
+
+    def initialize(request)
+      @request = request
+    end
+
+    def allows_http?
+      !requires_ssl?
+    end
+
+    def requires_ssl?
+      unattended_template_path? ? unattended_ssl? : !path_allows_http?
```

Comment:
Bike shedding: why `unattended_template_path` and then `unattended_ssl?` Should it be `unattended_path?` for consistency?